aea/package_manager/base.py

```diff
@@ -181,6 +180,30 @@ def check_dependencies(
                 )
         return mismatches
 
+    def is_dependencies_hashes_match(
+        self, package_id: PackageId, configuration_obj: PackageConfiguration
+    ) -> bool:
+        """Check dependecies hashes match and print errors"""
+        dependencies_with_mismatched_hashes = self.check_dependencies(
+            configuration=configuration_obj,
+        )
+        if len(dependencies_with_mismatched_hashes) > 0:
+            for dep, failure in dependencies_with_mismatched_hashes:
```

Comment:
the if check is not necessary here, can iterate an empty list

then `return bool(dependencies_with_mismatched_hashes)` at the end

Reply:
brilliant. done